tslearn/docs/examples/plot_sax_mindist_knn.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+"""
+1-NN with SAX + MINDIST
+=======================
+
+This example presents a comparison performs kNN with k=1, using MINDIST from
+the original paper as a distance function, on SAX transformations using
+different benchmark datasets. It compares its accuracy on a test set with
+the accuracy of kNN (k=1) using euclidean distance on the raw data.
+"""
+
+# Author: Gilles Vandewiele
+# License: BSD 3 clause
+
+import numpy
+import matplotlib.pyplot as plt
+import time
+
+from tslearn.datasets import UCR_UEA_datasets
+from tslearn.preprocessing import TimeSeriesScalerMeanVariance
+from tslearn.neighbors import KNeighborsTimeSeriesClassifier
+from tslearn.piecewise import PiecewiseAggregateApproximation, \
+    SymbolicAggregateApproximation, \
+    OneD_SymbolicAggregateApproximation
+
+from sklearn.pipeline import Pipeline
+from sklearn.metrics import pairwise_distances, accuracy_score, \
+    confusion_matrix
+from sklearn.preprocessing import FunctionTransformer
+from sklearn.neighbors import KNeighborsClassifier
+
+from scipy.stats import norm
+
+import warnings
+warnings.filterwarnings('ignore')
+
+
+def print_table(accuracies, times):
+    header_str = '|'
+    header_str += '{:^20}|'.format('dataset')
+    columns = ['sax error', 'sax time', 'eucl error', 'eucl time']
+    for col in columns:
+        header_str += '{:^12}|'.format(col)
+    print(header_str)
+    print('-'*(len(columns) * 13 + 22))
+
+    for dataset in accuracies:
+        acc_sax, acc_euclidean = accuracies[dataset]
+        time_sax, time_euclidean = times[dataset]
+        sax_error = numpy.around(1 - acc_sax, 5)
+        eucl_error = numpy.around(1 - acc_euclidean, 5)
+        time_sax = numpy.around(time_sax, 5)
+        time_euclidean = numpy.around(time_euclidean, 5)
+        s = '|'
+        s += '{:>20}|'.format(dataset)
+        s += '{:>12}|'.format(sax_error)
+        s += '{:>12}|'.format(time_sax)
+        s += '{:>12}|'.format(eucl_error)
+        s += '{:>12}|'.format(time_euclidean)
+        print(s.strip())
+
+    print('-'*(len(columns) * 13 + 22))
+
+
+numpy.random.seed(0)
+# Generate a random walk time series
+data_loader = UCR_UEA_datasets()
+datasets = [
+    ('SyntheticControl', 16),
+    ('GunPoint', 64),
+    ('FaceFour', 128),
+    ('Lightning2', 256),
+    ('Lightning7', 128),
+    ('ECG200', 32),
+    ('Plane', 64),
+    ('Car', 256),
+    ('Beef', 128),
+    ('Coffee', 128),
+    ('OliveOil', 256)
+]
+
+accuracies = {}
+times = {}
+for dataset, w in datasets:
+    X_train, y_train, X_test, y_test = data_loader.load_dataset(dataset)
+    metric_params = {'n_segments': w, 'alphabet_size_avg': 10}
+    knn = KNeighborsTimeSeriesClassifier(n_neighbors=1, metric='sax',
```

Comment:
You could probably define the classifiers outside the loop.

Reply:
I have made this change, but I had to do the following in order to get the same results as the previous version:
```python3
knn_sax = clone(knn_sax).set_params(metric_params=metric_params)
```

So not sure whether this is actually better than just re-instantiating the model in every loop.